server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala

```diff
@@ -306,18 +306,33 @@ class Serializer @Inject() (mailboxIdFactory: MailboxId.Factory) {
             "subParts" -> part.subParts.map(list => list.map(writes)))))
     }
 
-  private def emailWritesWithPropertyFilter(properties: Properties)(implicit partsWrites: Writes[EmailBodyPart]): Writes[Email] = {
+  private def emailWritesWithPropertyFilter(properties: Properties)(implicit partsWrites: Writes[EmailBodyPart]): Writes[EmailView] = {
     implicit val emailMetadataWrites: Writes[EmailMetadata] = Json.writes[EmailMetadata]
     implicit val emailHeadersWrites: Writes[EmailHeaders] = Json.writes[EmailHeaders]
     implicit val emailBodyWrites: Writes[EmailBody] = Json.writes[EmailBody]
 
-    val emailWrites: OWrites[Email] = (JsPath.write[EmailMetadata] and
+    val emailFullViewWrites: OWrites[EmailFullView] = (JsPath.write[EmailMetadata] and
         JsPath.write[EmailHeaders] and
-        JsPath.write[EmailBody]) (unlift(Email.unapply))
+        JsPath.write[EmailBody]) (unlift(EmailFullView.unapply))
 
-    emailWrites.transform(obj => properties.filter(obj))
+    val emailHeaderViewWrites: OWrites[EmailHeaderView] = (JsPath.write[EmailMetadata] and
+        JsPath.write[EmailHeaders]) (unlift(EmailHeaderView.unapply))
+
+    val emailMetadataViewWrites: OWrites[EmailMetadataView] = view => Json.toJson(view.metadata) match {
```

Comment:
you can use `Json.toJsObject `instead of `Json.toJson` this way you don't have to do this pattern matching and throwing.

Note that you need an `OWrites` and not a `Writes` for the metadata if it is not already the case

Reply:
Cool, huge thanks, exactly what I was looking for!

Reply:
But the issue is that an explicit OWrites is then needed for EmailMetadata ^^ (no free lunch)